cli/src/table.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt::Display;
+
+const PADDING: usize = 3;
+pub struct Column<T> {
+    header: &'static str,
+    value: Box<dyn Fn(&T) -> String>,
+}
+
+pub struct Table<'a, T> {
+    pub cols: Vec<Column<T>>,
+    pub data: &'a [T],
+}
+
+impl<T> Column<T> {
+    pub fn new(header: &'static str, value: Box<dyn Fn(&T) -> String>) -> Column<T> {
+        Column { header, value }
+    }
+
+    fn width(&self, rows: &[T]) -> usize {
+        match rows.iter().map(|t| (self.value)(t).len()).max() {
+            Some(width) => self.header.len().max(width) + PADDING,
+            None => self.header.len() + PADDING,
+        }
```

Comment:
nit, take it or leave it: this could also be expressed as
```suggestion
        let width = rows.iter().map(|t| (self.value)(t).len()).max().unwrap_or(0);
        self.header.len().max(width) + PADDING
```
which doesn't duplicate the math; theoretically this seems a bit nicer to me in case we ever want to add something else to the width calculation. in practice it's not a huge deal to duplicate this, though.